studiopress/genesis-custom-blocks
# Add the 'Block Settings' panel

#### Changes
* Uses Rob's markup and CSS
* Adds a 'Block Settings' panel:
![hetb-lock-pa](https://user-images.githubusercontent.com/4063887/99037617-a8d05d00-2549-11eb-9850-37a62353d40e.gif)


#### Testing instructions
1. `composer install && npm ci && npm run dev`
1. `wp post create --post_type='genesis_custom_block' --post_title='Test UI' --post_excerpt='' --post_content='{"genesis-custom-blocks\/test-ui":{"name":"test-ui","title":"Test UI","excluded":[],"icon":"genesis_custom_blocks","category":{"slug":"common","title":"Common Blocks","icon":null},"keywords":["baz"],"fields":{"text":{"name":"text","label":"Text","control":"text","type":"string","order":0,"location":"editor","width":"100","help":"","default":"","placeholder":"","maxlength":null}}}}'`
1. It should say something like `Success! Created post 8`
1. Go to the URL for that post: https://your-wp-local/wp-admin/post.php?post=8&action=edit
1. Expected: It should initially look like this:
![initial-display](https://user-images.githubusercontent.com/4063887/99107736-e9f95900-25ab-11eb-83c3-33ad33922bc1.png)

1. Expected: you can edit, undo, and save the 'Slug', 'Icon', 'Category', and 'Keywords', and 'Post Types': 
![het-block-tess](https://user-images.githubusercontent.com/4063887/99108191-9a675d00-25ac-11eb-8d80-74f24b896e7a.gif)
1. The scope of this PR is mainly the 'Block Settings' panel. There is still a lot of work needed in the editor.
